scripts/consensus_merge.py

```diff
@@ -0,0 +1,636 @@
+            DP_list.append(str(targ_sample.DP))
+
+        GT = stringify(GT_list, FORMAT_JOIN)
+        AD = stringify(AD_list, FORMAT_JOIN)
+        AF = stringify(AF_list, FORMAT_JOIN)
+        DP = stringify(DP_list, FORMAT_JOIN)
+
+        output_record.samples[index]['GTC'] = GT
+        output_record.samples[index]['ADC'] = AD
+        output_record.samples[index]['AFC'] = AF
+        output_record.samples[index]['DPC'] = DP
+
+        consensus_gt, gt_tag = get_gt_consensus(GT_list)
+
+#        print(consensus_gt, tuple(consensus_gt.split('/')))
+        output_record.samples[index]['GT'] = tuple([int(i) for i in consensus_gt.split('/')])
+        output_record.samples[index]['GT_STATUS'] = gt_tag
+        output_record.samples[index]['AD'] = get_ad_consensus(AD_list)
+        output_record.samples[index]['AF'] = get_af_consensus(AF_list)
+        output_record.samples[index]['DP'] = get_dp_consensus(DP_list)
+
+    chrom = single_caller_variants[0].record.chrom
+    pos = single_caller_variants[0].record.pos
+    mapq, mq0 = get_mapq(normal_cram, chrom, pos)
+
+    output_record.info['MQ'] = mapq
+    output_record.info['MQ0'] = mq0
+    output_record.info['CAL'] = ','.join([c for c in CALLER_NAMES if c.lower() in variant_lookup])
+
+    if hotspot:
+        output_record.info['Hotspot'] = True
+
+    output_record.filter.add('PASS')
+
+    output_vcf.write(output_record)
+
+def build_output_name(inpath, outbase):
+    """Builds an VCF.GZ output filename based on the input (VCF/VCF.GZ) name and given output basename
+       Args:
+           inpath (str): Path to the input VCF(.GZ) file
+           outbase (str): Given output basename string
+       Return:
+           str: Filename in the format <output_basename>.<input_nameroot>.consensus.vcf.gz
+    """
+    basename_split = os.path.split(inpath)[-1].split('.')
+    return '.'.join([outbase, basename_split[0] + '.consensus.vcf.gz'])
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--strelka2_vcf', 
+                        help='Strelka2 VCF with MNPs fixed')
+    parser.add_argument('--mutect2_vcf',
+                        help='Mutect2 VCF')
+    parser.add_argument('--lancet_vcf', 
+                        help='Lancet VCF')
+    parser.add_argument('--vardict_vcf',
+                        help='VarDict VCF')
+    parser.add_argument('--cram', help='CRAM file')
+    parser.add_argument('--output_basename',
+                        help='String to use as basename for output file')
+
+    args = parser.parse_args()
+
+    # Create VCF and CRAM pysam objects
+    strelka2_vcf = pysam.VariantFile(args.strelka2_vcf, 'r')
+    mutect2_vcf = pysam.VariantFile(args.mutect2_vcf, 'r')
+    lancet_vcf = pysam.VariantFile(args.lancet_vcf, 'r')
+    vardict_vcf = pysam.VariantFile(args.vardict_vcf, 'r')
+
+    normal_cram = pysam.AlignmentFile(args.cram, 'rc') #, reference_filename="/home/ubuntu/volume/ref/Homo_sapiens_assembly38.fasta.fai")
```

Comment:
Yes, that's what I was experiencing! I convinced myself there was something wrong with that cram because it actually worked up to ~chr 10. Are you using the one I pointed to in the PR? Is there any chance you have another hg38 one handy and could try with that? I feel it shouldn't be necessary to provide the reference, and if it's just this cram that's broken ok, but if not I'd better fix it.

Reply:
Hmmm, try downloading from here: https://cavatica.sbgenomics.com/u/zhangb1/kf-somatic-tools-test/files/#q?search=Homo_sapiens_assembly38.fasta
Specifically files https://cavatica.sbgenomics.com/u/zhangb1/kf-somatic-tools-test/files/5b27fd8fec70495356f7da6a/ and https://cavatica.sbgenomics.com/u/zhangb1/kf-somatic-tools-test/files/5b27fd8fec70495356f7da30/

Reply:
I tried grabbing a different cram from Cavatica, but had the same problem where `samtools index` doesn't generate a functional index past chr10, but you can slice in past that point by passing the fasta or fai explicitly. Downloaded index from Cavatica instead of trying to create locally, but same problem. 

To resolve this I have added a `reference` option to the script, but I feel like this requirement is not normal. If you have time to check, if you just download an hg38-aligned cram from Cavatica, are you able to index it locally with samtools and then slice in at chr11 or later?

Reply:
I think I finally get it now. I was working with crams and crai that I downloaded from cavatica. You were using samtools to index the cram you downloaded first, then samtools was giving you an error. I see that too. However, oddly enough, pysam throws me no errors if I grab something from chromosome 12 if either cram index is present, as long as I give it a fasta file, but a cram index must exist. However, for the samtools issue, I was able to generate an index error-free if I used version 1.9, instead of the installed 1.8.
So in short:
1. To generate a cram index locally rather than use the one on cavatica, use samtools 1.9
2. This works:
```
test = pysam.AlignmentFile("4900e964-eb56-451b-a04d-f24155b8b7ef.cram", "rc", reference_filename="Homo_sapiens_assembly38.fasta")
check = test.fetch("chr12", 125000000, 125000001)
read = next(check)
print(read)
```
3. This does not:
```
test = pysam.AlignmentFile("4900e964-eb56-451b-a04d-f24155b8b7ef.cram", "rc")
check = test.fetch("chr12", 125000000, 125000001)
read = next(check)
[W::find_file_url] Failed to open reference "https://www.ebi.ac.uk/ena/cram/md5/96e414eace405d8c27a6d35ba19df56f": Protocol not supported
[E::cram_get_ref] Failed to populate reference for id 11
[E::cram_decode_slice] Unable to fetch reference #11 124973272..125019780

[E::cram_next_slice] Failure to decode slice
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pysam/libcalignmentfile.pyx", line 2086, in pysam.libcalignmentfile.IteratorRowRegion.__next__
OSError: truncated file
```
Hope that helps!
P.S. Yeah, I agree that needing a reference normally isn't needed, but that's usually for bam input. Form cram input, the accompanying reference information is generated as-needed, so the reference is required. It might be because our headers for the reference resource don't point to that remote server, rather the the path that the reference was at that time. So perhaps other under circumstances, had we pointed to a URI that has the reference, we might not need to have that at run time.

Reply:
Thanks for investigating! That sounds similar to my experience, and what you were saying about the remote reference sounds like a plausible explanation. It would be great if that problem got characterized and solved, but the reference option I added should provide a workaround as far as this script goes. 

From your perspective, is anything blocking merge at this point?